app/javascript/src/ChangePassword/components/ChangePasswordHooks.jsx

```diff
@@ -1,87 +1,95 @@
+// @flow
+
 import { useState } from 'react'
-import { validateForm, validateSingleField } from 'LoginPage/utils/formValidation'
+import validate from 'validate.js' // TODO: including this since validateSingleField and validateForm from LoginPage/utils/formValidation screams for a refactor
+
+type PasswordField = 'user[password]'
+type PasswordConfirmationField = 'user[password_confirmation]'
+type FieldName = PasswordField | PasswordConfirmationField
+type Constraints = {[FieldName]: {[string]: {}}}
+type FieldState = {[FieldName]: string}
+type ValidationErrors = {[FieldName]: string[]}
+type FieldErrorsState = ValidationErrors
+
+const PASSWORD: PasswordField = 'user[password]'
```

Comment:
I think assigning types to constants is a bit excessive! `PASSWORD` exists just to avoid repetition, types should be used with params and variables.

Reply:
Well, in Haskell or Elm, when you manage specific strings, those are recommended to be types, not strings. Like in this case, it's not any string the field name, it's a specific string. Any other string won't work, and will crash the app when the form is submited.

Reply:
I know what you mean and usually Flow does it for you, it's implied. CONSTANTS (upper case) are meant to avoid repetition, to declare absolute and generic content, they are type agnostic. Types are used to watch for inputs, unexpected inputs. You wouldn't declare:
```
const PORT: number = 3000
```
It doesn't matter what type it is, its value is absolute and unchangeable. What you would do is:
```
function startServer (port: number); // any number will suffice, will throw if PORT wouldn't be a number
function startServer (port: 3000); // it can only be 3000, will throw if PORT is other number than 3000
```
Defining a custom type is redundant:
```
type Port = 3000
const PORT: Port = 3000
```

Reply:
Well... Flow actually doesn't do it for you. But if we follow the old C convention of symbolic constants, it makes sense. This is defined in our file, we don't get it from external input. Will change